src/query/storage/prometheus/prometheus_storage.go

```diff
@@ -154,7 +154,9 @@ func (q *querier) Select(
 		return promstorage.ErrSeriesSet(err)
 	}
 
-	*resultMetadataPtr = result.Metadata
+	// XXX(esmet): This is likely a race condition.
```

Comment:
Is this a TODO?

Reply:
This is actually addressed in another PR https://github.com/m3db/m3/pull/3982

Reply:
#3982 landed 👍